docs/electrical/embedded/styleguide.md

```diff
@@ -1,3 +1,136 @@
+
+
+### Comment Rules
+1. Both single (//) and multi (/*) comment styles are considered acceptable, and should be used as applicable. When multi-line comments are used, all lines should begin with an asterisk (\*)
+2. Multi-line comments should maintain a level of indentation with the leading asterisks vertical to each other, as shown below.
+3. One leading space should be put between the comment text and the comment symbol.
+4. If placed at the end of a line of code, all comments should be vertically aligned.
+```c
+/* comment text
+ * comment text
+ * comment text
+ */
+
+ // comment text
+```
+
+
+5. All comments should be written in coherent, complete sentences.
+6. Assume a competentcy with the language from a reader, and avoid explaining the obvious. (*You aren't trying to hit a comment count for an assignment, you're trying to actually be helpful to yourself and others in the future*)
+7. Single line comments should be used where the function of a single line may be ambiguous.
+
+*TODO: Consider whether it's necessary to require conformance to automatic documentation generation standards*
+
+#### Comment Locations
+1. Functions should be preceeded by a short comment block explaining the relevant values (inputs, outputs) and an extremely brief description of the purpose of the function.
+```c title="Example"
+/* [sum]
+ * Returns the sum of two numbers
+ *
+ * Input: int A, int B
+ * Output: int C
+ */
+
+int sum(int A, int B) {
+    int c;
+    c = a + b;
+    return c;
+}
+```
+
+### White Space Rules
+1. The keywords ```if, while, for, switch,``` and ```return``` should be followed by a single space if there is more text on the same line.
+2. Unless specified by another whitespace rule, all operators should be surrounded on either side by a single space.
+3. Unary operators should not have a space between the operator and the operand.
+4. Structure pointer and member operators should have no whitespace on either side.
+5. Parentheses, braces, and brackets should not have an adjacent space on the interior side.
+6. Excepting the end of a line, separating commas should should be followed by one space.
+7. Semicolons should follow a preceeding statement without a space, but should always be followed by a space.
+8. Use whitespace to separate blocks of code from each other in a logical fashion. (William's note: to please my OCD use either 1 or 3 lines of separation depending on context, don't use two :) )
+
+#### Pointer (* and &) Operators
+1. When used with a type, such as in the case of ```pointerType name```, the pointer operator * should be appended to the type name (i.e. ```type* name```).
+2. Otherwise both operators should be written without a space on the operand side.
+
+#### Alignment Rules
+1. TODO: Consider whether to always left align the # of a preprocessor directive, regardless of the indentation of the line
```

Comment:
It is commonly encouraged to do so to make preprocessor stuff extremely obvious. However, I understand arguments against at times. Probably warrants discussion.

Reply:
My first instinct would be to lean towards left alignment for sake of being obvious. However, I don't know enough about the cases against to feel confident in a general judgement.

Reply:
Left alignment is what is done in almost every style guide. I'd say just stick with that. The arguments against it are not (IMO) worth worrying about.